core/src/main/java/com/google/zxing/BinaryBitmap.java

```diff
@@ -138,6 +138,16 @@ public BinaryBitmap rotateCounterClockwise45() {
     return new BinaryBitmap(binarizer.createBinarizer(newSource));
   }
 
+  /**
+   * Returns a new object with negative image data.
+   *
+   * @return An inverted version of this object.
+   */
+  public BinaryBitmap invert() {
+    LuminanceSource newSource = binarizer.getLuminanceSource().invert();
+    return new BinaryBitmap(binarizer.createBinarizer(newSource));
```

Comment:
I think it might be more efficient to just add a flip() method for the whole BitMatrix, and call it here? This is sound too as it reuses the (unused!) LuminanceSource.invert() method but has to binarize the result again

Reply:
Funny that you point that out. Inverting the BitMatrix itself was actually my first implementation.

I switched to the existing invert method because I try to reuse existing code as much as possible, when contributing to a foreign code base. I'll gladly switch back to the other implementation.

The test failure indicates a small leftover of https://github.com/zxing/zxing/pull/1372 when separating the code of both PRs.
I'll update the PR soon.

Reply:
OK I may eventually just convert this to flipping the bits directly, but I don't mind as a first pass.